seatunnel-connectors-v2/connector-elasticsearch/src/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/client/EsRestClient.java

```diff
@@ -0,0 +1,120 @@
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpHost;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.util.EntityUtils;
+import org.apache.seatunnel.connectors.seatunnel.elasticsearch.dto.BulkResponse;
+import org.apache.seatunnel.connectors.seatunnel.elasticsearch.exception.BulkElasticsearchException;
+import org.apache.seatunnel.connectors.seatunnel.elasticsearch.exception.GetElasticsearchVersionException;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestClientBuilder;
+
+import java.io.IOException;
+import java.util.List;
+
+public class EsRestClient {
+
+    private static EsRestClient esRestClient;
+    private static RestClient restClient;
+
+    private EsRestClient() {
+
+    }
+
+    private static RestClientBuilder getRestClientBuilder(List<String> hosts, String username, String password) {
+        HttpHost[] httpHosts = new HttpHost[hosts.size()];
+        for (int i = 0; i < hosts.size(); i++) {
+            String[] hostInfo = hosts.get(i).replace("http://", "").split(":");
+            httpHosts[i] = new HttpHost(hostInfo[0], Integer.parseInt(hostInfo[1]));
+        }
+
+        RestClientBuilder builder = RestClient.builder(httpHosts)
+                .setRequestConfigCallback(requestConfigBuilder -> requestConfigBuilder
+                        .setConnectionRequestTimeout(10 * 1000)
+                        .setSocketTimeout(5 * 60 * 1000));
+
+        if (StringUtils.isNotEmpty(username)) {
+            CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+            credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(username, password));
+            builder.setHttpClientConfigCallback(httpAsyncClientBuilder -> httpAsyncClientBuilder.setDefaultCredentialsProvider(credentialsProvider));
+        }
+        return builder;
+    }
+
+    public static EsRestClient getInstance(List<String> hosts, String username, String password) {
+        if (restClient == null) {
+            RestClientBuilder restClientBuilder = getRestClientBuilder(hosts, username, password);
+            restClient = restClientBuilder.build();
+            esRestClient = new EsRestClient();
+        }
+        return esRestClient;
+    }
+
+    public BulkResponse bulk(String requestBody) {
+        Request request = new Request("POST", "_bulk");
+        request.setJsonEntity(requestBody);
+        try {
+            Response response = restClient.performRequest(request);
+            if (response.getStatusLine().getStatusCode() == 200) {
```

Comment:
It's better use` HttpStatus` judge.

Reply:
And maybe be NPE?

Reply:
I have revised  it according to your suggestion.